### PR TITLE
Trying to use build matrix

### PR DESCRIPTION
**Pull request by pariterre**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by pariterre** [comment]
@isuruf 
Thanks for pointing the matrix stuff. In order to make everything right, I started back to the root of the problem, RBDL (this relate to the PR: https://github.com/conda-forge/staged-recipes/pull/11627). This can be build using either Eigen3 or CasADi. I've added the both in the build matrix for linux and osx (as casadi won't compile for windows) and only eigen for windows.

This seems to work fine. I have 3 questions though. 
First, I've tried to make linter happy, however I don't find any better way to write the host and run in meta.yaml than: 
```
  host:
    - {{ backend }}
```
It works to compile, but linter doesn't understand. If I write:
```
  host:
    - {{ 'backend' }}
```
Then the linter is happy, but this translate to backend=casadi instead of solely casadi

Second question is what a lambda user will have to do to install rbdl using one backend or the other?
Will `conda install rbdl -cconda-forge` work? If so, which of the backend is the default? To use the other one will this command do the trick? `conda install rbdl=casadi -cconda-forge`?

Finally, when this is done, I suggest to remove the package rbdl-casadi which is sole purpose is to offer the casadi backend of the same package, but I am not sure how. 

Thanks for your time!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,9 +24,9 @@ requirements:
     - cmake
     - ninja
   host:
-    - eigen
+    - {{ backend }}
```
Comment (isuruf) [inline recipe/meta.yaml:27]: Just to make the linter happy
```suggestion
    - {{ backend|default("casadi") }}
```

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+backend:
+  - 'casadi'  # [linux or osx]
+  - 'eigen'
+  
+backend_name_in_build:
+  - 'Casadi'  # [linux or osx]
+  - 'Eigen3'
+  
+zip_keys:
+  - backend
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:10]:
```suggestion
-
  - backend
```

**Comment by isuruf** [comment]
You can set the build string to be `{{ backend }}_h{{ PKG_HASH }}_{{ build_num }}` and a user can select by doing `rbdl=*=casadi*`

**Comment by pariterre** [comment]
@conda-forge-admin, please rerender

**Comment by pariterre** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
You need to add a `track_features` to make sure `casadi` backend is not installed by default.

**Comment by pariterre** [comment]
@isuruf 
I've had a look that, but I don't really understand what it does from the doc. Should this be something like:
```
build:
  track_features:
    - casadi
```
?

**Comment by isuruf** [comment]
```
build:
  track_features:
    - casadi             # [backend == "casadi"]
```

This makes the priority of casadi backend package lower so that when a user requests `rbdl` they'll get the eigen backend unless `casadi` is specifically requested.

**Pull request merged**